### PR TITLE
Many PEP8 fixes to the stats subpackage

### PR DESCRIPTION
**Pull request by larrybradley**
A myriad of PEP8 fixes.  Also removed several unused imports.

### CODE REVIEW

**Comment by pllim** [comment]
Curious. Why didn't the Travis CI PEP8 test catch these?

**Comment by larrybradley** [comment]
@pllim It appears that the `travis-ci` PEP8 test doesn't actually test anything, e.g.:
https://travis-ci.org/astropy/astropy/jobs/136560092#L465

However, I wouldn't recommend turning it on because there are many PEP8 issues throughout the package.

cc: @astrofrog, @bsipocz

**Comment by pllim** [comment]
That is definitely weird. When I use the PEP8 test incantation from Astropy package template, it wouldn't stop complaining about my codes!

**Comment by larrybradley** [comment]
The latest package template `.travis.yml` doesn't have any PEP8 tests.

**Comment by pllim** [comment]
Oh wait... maybe I didn't copy it from the template, but rather from here! See [mine](https://github.com/spacetelescope/synphot_refactor/blob/master/.travis.yml). It was definitely [complaining about all sorts of PEP8](https://travis-ci.org/spacetelescope/synphot_refactor/jobs/132950080).

**Comment by astrofrog** [comment]
This will need rebasing - I'm milestoning as 1.2.1 since I'd like to get 1.2 out and this is not critical

**Comment by bsipocz** [comment]
@pllim @larrybradley - pep8 is run on travis only with a very small subset of the errors (they are listed in setup.cfg). As always it's difficult to extend that list.

Also whenever you run pep8 in your checked out repo, unless you force it otherwise, it will only report the errors listed in the config file.

**Comment by larrybradley** [comment]
@astrofrog  Rebased.

@bsipocz Thanks for the info.  I agree it's best to only automatically check for a subset of errors and those in `setup.cfg` are the most important for consistency.

File: astropy/stats/tests/test_jackknife.py [hunk 1]
```diff
@@ -30,7 +30,6 @@ def test_jackknife_resampling():
 def test_jackknife_stats():
     # Test from the third example of Ref.[3]
     data = np.array((115, 170, 142, 138, 280, 470, 480, 141, 390))
-    resamples = jackknife_resampling(data)
```
Comment (astrofrog) [inline astropy/stats/tests/test_jackknife.py]: Why remove this line? Maybe just remove the ``resamples = `` part?

Comment (astrofrog) [inline astropy/stats/tests/test_jackknife.py]: Ah no ignore me

File: astropy/stats/tests/test_funcs.py [hunk 2]
```diff
@@ -465,11 +487,11 @@ def test_poisson_conf_frequentist_confidence_gehrels_2sigma():
...
                     (9, 2, 4.046, 17.27),
                     (10, 2, 4.719, 18.58)])
-    assert_allclose(funcs.poisson_conf_interval(nlh[:,0],
-                        sigma=2,
-                        interval='frequentist-confidence').T,
-                    nlh[:,2:],
-                    rtol=0.01)
+    assert_allclose(
+        funcs.poisson_conf_interval(nlh[:, 0], sigma=2,
+                                    interval='frequentist-confidence').T,
```
Comment (astrofrog) [inline astropy/stats/tests/test_funcs.py]: This doesn't look quite right, it should be similar to before but de-denting the sigma and interval lines

File: astropy/stats/tests/test_funcs.py [hunk 3]
```diff
@@ -440,10 +461,11 @@ def test_poisson_conf_frequentist_confidence_gehrels():
                     (9, 6.057, 13.11),
                     (10, 6.891, 14.27),
                     ])
-    assert_allclose(funcs.poisson_conf_interval(nlh[:,0],
-                        interval='frequentist-confidence'),
```
Comment (astrofrog) [inline astropy/stats/tests/test_funcs.py]: Dedent this line and leave the rest the same

File: astropy/stats/tests/test_funcs.py (hunk 2 above)
Comment (larrybradley) [inline astropy/stats/tests/test_funcs.py]: No, de-denting `sigma` and `interval` is wrong.  They are keywords to `poisson_conf_interval`, so they would need to be indented *more* to line up under `nlh`.  But then the `interval` keyword greatly overruns the 80-column limit.

File: astropy/stats/tests/test_funcs.py (hunk 3 above)
Comment (larrybradley) [inline astropy/stats/tests/test_funcs.py]: Same here.

File: astropy/stats/tests/test_funcs.py (hunk 2 above)
Comment (bsipocz) [inline astropy/stats/tests/test_funcs.py]: what about leaving ``funcs.poisson...`` in the first line and  adding the new line before ``nlh``? Then ``interval`` can be lined up under it and fit into the 80.

Comment (larrybradley) [inline astropy/stats/tests/test_funcs.py]: Yes, that's the other option.  @astrofrog?

Comment (astrofrog) [inline astropy/stats/tests/test_funcs.py]: Ok, re-reading the code I realize this is actually fine, so I'll just go ahead and merge.

**Pull request merged**